### ISSUE TITLE
Incorrect "Use of unassigned local variable" error

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1667]
The code below doesn't compile because of an erroneous "Use of unassigned local variable 'a'" error.
```cs
		public static bool Any<TCollection, TSource, TPredicate>(TCollection items, TPredicate predicate)
			where TCollection : concrete, IEnumerable<TSource>
			where TPredicate : delegate bool(TSource)
		{
			for (var it in items)
				if (predicate(it))
					return true;

			return false;
		}

		public static void Main()
		{
			List<int> a;
			if (true)
			{
				a = new List<int>();
				bool b = Any(a, (i) => a.Contains(i));
			}
		}
```
Tested with: https://github.com/beefytech/Beef/commit/1806cb923bb319ebc18559e1bffa87990525d6a4

**Comment by bfiete** [comment]
Fixed at 37390205048e0bcdeeffaa19a4bd98e534206a88

**Comment by disarray2077** [comment]
@bfiete The fix is causing a crash when I open my actual project, I will try to isolate it but here's a screenshot that can help:
![image](https://user-images.githubusercontent.com/86157825/182034040-a1c3f6d4-2fe6-49f7-bf1a-7aa2a2886076.png)

**Comment by bfiete** [comment]
Oops.
Fixed at 39595bf38714b7e4ea1c7aca89d2234f49ddfd5c
